.github/workflows/tests.yml

```diff
@@ -6,36 +6,42 @@ on:
 
 jobs:
   test-minimal-deps:
-    runs-on: ubuntu-latest
-
+    strategy:
+      matrix:
+        platform: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: ['3.8', '3.9', '3.10', '3.11']
+    runs-on: ${{ matrix.platform }}
     steps:
-    - uses: actions/checkout@v3
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v4
-      with:
-        python-version: ${{ matrix.python-version }}
-    - name: Install poetry
-      run: pipx install poetry
-    - name: Install poetry dependencies
-      run: |
-        poetry install --with dev || echo "No dev dependency group"
-    - name: Run poetry tests (minimal-deps)
-      run: poetry run pytest tests/0_core/
+      - uses: actions/checkout@v3
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install poetry
+        run: pipx install poetry
+      - name: Install poetry dependencies
+        run: |
+          poetry install --with dev || echo "No dev dependency group"
+      - name: Run poetry tests (minimal-deps)
+        run: poetry run pytest tests/0_core/
 
   test-full-deps:
     needs: test-minimal-deps
-    runs-on: ubuntu-latest
     strategy:
       matrix:
         platform: [ubuntu-latest, macos-latest, windows-latest]
         python-version: ['3.8', '3.9', '3.10', '3.11']
-
+    runs-on: ${{ matrix.platform }}
```

Comment:
I realised that even without this tests seem to run on all platforms. Not sure why... Might consider removing this again.

Reply:
Good point. Could we move the matrix-testing to a new pull-request (and merge this one)?